### PR TITLE
Update to RxJS 6

### PR DESCRIPTION
**Pull request by Roadirsh**
Correct new import

**Summary**

This PR fixes/implements the following **bugs/features**

- [ https://github.com/orizens/ngx-infinite-scroll/issues/271] #271 Update for RXJS 6
- It is a breaking change

With this pull request we can keep the package up to date

### CODE REVIEW

**Comment by orizens** [comment]
@Roadirsh 
this looks good. 
can you run the tests and make to update test files?

**Comment by Roadirsh** [comment]
@orizens Ok now the test passed but you will need to create a new version 0.9 for the ones that want to use it with Angular 6, thanks for your package !

**Comment by kuncevic** [comment]
@orizens when the new version will be published?

**Comment by jdwholmes** [comment]
@orizens Do we have a date for this?

**Comment by orizens** [comment]
soon

File: package.json [hunk 1]
```diff
@@ -29,26 +28,30 @@
...
-  "keywords": ["angular", "javascript", "typescript"],
+  "keywords": [
+    "angular",
+    "javascript",
+    "typescript"
+  ],
   "license": "MIT",
   "peerDependencies": {
     "@angular/common": ">= 4.0.0",
     "@angular/core": ">= 4.0.0"
```
Comment (orizens) [inline package.json]: what about this packages?
do we want to require v 6.0.0?

File: tests/services/scroll-resolver.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as Models from '../../src/models';
 import * as ScrollResolver from '../../src/services/scroll-resolver';
-
+import {} from 'jasmine';
```
Comment (orizens) [inline tests/services/scroll-resolver.spec.ts]: why is this empty?

File: tsconfig.json [hunk 3]
```diff
@@ -17,8 +17,7 @@
         "target": "es5",
         "skipLibCheck": true,
         "types": [
-            "jasmine",
-            "node"
+            "jasmine"
```
Comment (orizens) [inline tsconfig.json]: why "node" was removed?

File: src/modules/infinite-scroll.directive.ts [hunk 4]
```diff
@@ -1,104 +1,105 @@
...
+    EventEmitter,
+    Input,
+    NgZone,
+    OnChanges,
+    OnDestroy,
+    Output,
+    SimpleChanges
 } from '@angular/core';
-import { Subscription } from 'rxjs/Subscription';
+import {Subscription} from 'rxjs/index';
```
Comment (MrBlaise) [inline src/modules/infinite-scroll.directive.ts]: Why is this imported from `rxjs/index` and not `rxjs`?

File: src/modules/infinite-scroll.directive.ts [hunk 5]
```diff
@@ -1,104 +1,105 @@
...
-  @Input() scrollWindow: boolean = true;
-  @Input() immediateCheck: boolean = false;
-  @Input() horizontal: boolean = false;
-  @Input() alwaysCallback: boolean = false;
-  @Input() fromRoot: boolean = false;
+    @Input() infiniteScrollDistance: number = 2;
+    @Input() infiniteScrollUpDistance: number = 1.5;
+    @Input() infiniteScrollThrottle: number = 150;
+    @Input() infiniteScrollDisabled: boolean = false;
+    @Input() infiniteScrollContainer: any = null;
```
Comment (MrBlaise) [inline src/modules/infinite-scroll.directive.ts]: This pull request should only concern rxjs 6 and not change the coding style in this case the indentation, in my opinion.

**Comment by deeg** [comment]
@Roadirsh, will you be able to address the comments here? If not, no worries, I can create another Pull Request.

I am looking to help out to continue this moving and getting it merged as I'm selfishly looking for it :).

Please let me know if anyone needs anything!

**Comment by deeg** [comment]
I have tested with my Angular 6, rxjs 6 app and everything is working.

You can try out in your app by replacing in your package.json:

```"ngx-infinite-scroll": "deeg/ngx-infinite-scroll#rxjs6-dist"```

**Comment by Roadirsh** [comment]
@deeg Most of the answer of the comments are "Because my IDE told me too made this change" ahah. 

I have some time today, so I will try to made the changes in this PR anyway, at least for the beauty of coding.

File: src/services/scroll-register.ts [hunk 6]
```diff
@@ -1,128 +1,125 @@
-import { ElementRef } from "@angular/core";
-import { Observable } from "rxjs/Observable";
-import { of } from "rxjs/observable/of";
-import { fromEvent } from "rxjs/observable/fromEvent";
-import { map, mergeMap, tap, sampleTime, filter } from "rxjs/operators";
+import {fromEvent, Observable, of} from "rxjs/index";
```
Comment (MrBlaise) [inline src/services/scroll-register.ts]: From rxjs not rxjs/index

File: tests/services/scroll-register.spec.ts [hunk 7]
```diff
@@ -1,123 +1,119 @@
 import * as Models from '../../src/models';
-import { Observable } from 'rxjs/Observable';
-import {
-  async,
-  inject
-} from '@angular/core/testing';
+import {Observable} from 'rxjs/index';
```
Comment (MrBlaise) [inline tests/services/scroll-register.spec.ts]: From rxjs not rxjs/index

**Pull request closed without merging**